pimd/pim_igmp.c

```diff
@@ -469,6 +469,45 @@ static int igmp_v1_recv_report(struct igmp_sock *igmp, struct in_addr from,
 	return 0;
 }
 
+int pim_igmp_verify_header(struct ip *ip_hdr, size_t len, int igmp_msg_len,
```

Comment:
Let's make the return a bool and return true the packet header has been verified and false for not.  But as it stands -1 for  not verified and 0 for verified is an awful choice of return codes